docs/documentation-generation/docs-readme.md

```diff
@@ -7,54 +7,478 @@ slug: /docs-readme
+    { 
+      type: 'docs-readme'
+    }
   ]
 };
 ```
 
-Another option would be to use the `--docs` CLI flag as a part of the [build task](../config/cli.md#stencil-build), like so:
+## Generating README Files
+
+### Using the Build Task
 
+If your project has a `docs-readme` output target configured in your Stencil configuration file, the Stencil [build task](../config/cli.md#stencil-build) is all that's needed to generate README docs:
 ```bash
-stencil build --docs
+npx stencil build
+```
+If you're running the build task with the `--watch` flag, your project's README files will automatically update without requiring multiple explicit build commands:
+```bash
+npm stencil build --watch
 ```
 
-This will cause the Stencil compiler to perform a one-time generation of README
-files as a part of the build process.
-
-:::note
-If you don't add the `docs-readme` output target to your Stencil configuration, the `--docs` flag must be applied to regenerate documentation.
+:::info
+When running the build task with the `--dev` flag, README files will not be generated.
+This is to prevent unnecessary I/O operations during the development cycle.
 :::
 
-Alternatively, the [docs task](../config/cli.md#stencil-docs) can be used to perform a one time generation of the documentation:
+If you choose not to include a `docs-readme` output target in your Stencil configuration file, use the `--docs` CLI flag as a part of the build task:
+```bash
+npx stencil build --docs
+```
+
+This will cause the Stencil compiler to perform a one-time build of your entire project, including README files.
+
+### Using the Docs Task
+
+As an alternative to the build task, the [docs task](../config/cli.md#stencil-docs) can be used to perform a one time generation of the documentation:
 ```bash
-stencil docs
+npx stencil docs
+```
+Running `stencil docs` will generate documentation for [all documentation output targets](./01-overview.md), not just `docs-readme`.
+
+## README Sections
+
+Most generated markdown content will automatically be generated without requiring any additional configuration.
+Content is generated based on its Stencil component, rather than requiring you to configure multiple flags.
+Each section below describes the different types of content Stencil recognizes and will automatically generate. 
+
+### Custom Markdown Content
+
+Once you've generated a `readme.md` file, you can add your own markdown content to the file.
+You may add any content above the following comment in a component's `readme.md`:
+```
+Custom content goes here!
+<!-- Auto Generated Below -->
+```
+
+Any custom content placed above this comment will be persisted on subsequent builds of the README file.
+
+### Deprecation Notices
+
+A Stencil component may be marked as deprecated using the [JSDoc `@deprecated` tag](https://jsdoc.app/tags-deprecated).
+By placing `@deprecated` in a component's class-level JSDoc will cause the generated README to denote the component is deprecated.
```

Comment:
```suggestion
By placing `@deprecated` in a component's class-level JSDoc it will cause the generated README to denote the component as deprecated.
```